File: qc-core/src/main/java/com/eager/questioncloud/core/domain/cart/service/CartService.java
package com.eager.questioncloud.core.domain.cart.service;

import com.eager.questioncloud.core.domain.cart.dto.CartItemDetail;
import com.eager.questioncloud.core.domain.cart.implement.CartItemAppender;
import com.eager.questioncloud.core.domain.cart.implement.CartItemReader;
import com.eager.questioncloud.core.domain.cart.implement.CartItemRemover;
import com.eager.questioncloud.core.domain.cart.model.CartItem;
import com.eager.questioncloud.core.domain.question.implement.QuestionReader;
import com.eager.questioncloud.core.domain.question.model.Question;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class CartService {
    private final CartItemReader cartItemReader;
    private final CartItemAppender cartItemAppender;
    private final CartItemRemover cartItemRemover;
    private final QuestionReader questionReader;

    public List<CartItemDetail> getCartItems(Long userId) {
        return cartItemReader.getCartItems(userId);
    }

    public void appendCartItem(Long userId, Long questionId) {
        Question question = questionReader.getQuestion(questionId);
        CartItem cartItem = CartItem.create(userId, question.getId());
        cartItemAppender.append(cartItem);
    }

    public void removeCartItem(List<Long> ids, Long userId) {
        cartItemRemover.remove(ids, userId);
    }
}

File: qc-core/qc-cart/qc-cart-api/src/main/kotlin/com/eager/questioncloud/cart/service/CartService.kt
package com.eager.questioncloud.cart.service

import com.eager.questioncloud.cart.dto.CartItemDetail
import com.eager.questioncloud.cart.implement.CartItemAppender
import com.eager.questioncloud.cart.implement.CartItemDetailReader
import com.eager.questioncloud.cart.implement.CartItemRemover
import com.eager.questioncloud.cart.implement.CartItemValidator
import org.springframework.stereotype.Component

@Component
class CartService(
    private val cartItemValidator: CartItemValidator,
    private val cartItemAppender: CartItemAppender,
    private val cartItemDetailReader: CartItemDetailReader,
    private val cartItemRemover: CartItemRemover,
) {
    fun getCartItemDetails(userId: Long): List<CartItemDetail> {
        return cartItemDetailReader.getCartItemDetails(userId)
    }
    
    fun appendCartItem(userId: Long, questionId: Long) {
        cartItemValidator.validate(userId, questionId)
        cartItemAppender.append(userId, questionId)
    }
    
    fun removeCartItem(ids: List<Long>, userId: Long) {
        cartItemRemover.removeCartItem(ids, userId)
    }
}
